### PR TITLE
model: Add mDenseOn/mLateOn definitions

### PR DESCRIPTION
**Pull request by NohTow**
This is a PR to add the definition of the mLateOn/mDenseOn models, the multilingual extension of LateOn/DenseOn (#4545) trained with translate-train on eight target languages plus English.

Collection: https://huggingface.co/collections/lightonai/mdenseon-and-mlateon
Blogpost: https://huggingface.co/blog/lightonai/mDenseOn-mLateOn
Paper: https://arxiv.org/abs/2607.27178
Corresponding results PR: https://github.com/embeddings-benchmark/results/pull/653

A few notes:
- As with LateOn/DenseOn, the training data is a reproduction of the data we publicly release that is Apache 2.0 compatible. For the zero-shot computation, I listed as overlap what I find overlapping with the public data for fairness (same list as #4545 for the English seed, since the multilingual pre-training corpus is a translate-train extension of it).
- On top of the translated fine-tuning data, the models are fine-tuned on the training splits of MIRACL, MLDR, and the LateOn-Code data (which covers the MTEB code tasks), so those are listed in `training_datasets`. For CodeEditSearch, we built a training set from CommitPackFT and decontaminated it against the evaluation set (shared commit SHAs, exact normalized-text matches, and ≥50% 13-gram containment), but I still list it as overlap to be conservative.
- `languages` lists English plus the eight translate-train target languages (French, German, Italian, Spanish, Portuguese, Swedish, Norwegian, Arabic). The models transfer well beyond these (see the blog post), but I only listed the languages actually present in retrieval training.
- The backbone is [mmBERT-base](https://huggingface.co/jhu-clsp/mmBERT-base) (307M parameters, 8192 context).

cc @Samoed

### CODE REVIEW

File: mteb/models/model_implementations/pylate_models.py [hunk 1]
```diff
@@ -1020,3 +1004,108 @@ def _encode(
...
+    "ruby-Code",
+    "php-Code",
+]
+
+mdenseon_mlateon_citation = r"""@misc{sourty2026mdenseonmlateon,
+  title={{mDenseOn with the mLateOn}: Open Multilingual, Long-Context, and Code Retrieval Models},
+  author={Sourty, Raphael and Chaffin, Antoine and Moura Junior, Paulo Roberto and Chatelain, Amelie},
+  year={2026},
+  howpublished={\url{https://huggingface.co/blog/lightonai/mDenseOn-mLateOn}},
+}"""
```
Comment (Samoed) [inline mteb/models/model_implementations/pylate_models.py]: Maybe change to https://arxiv.org/abs/2607.27178?

**Review by Samoed — APPROVED** [review]

File: mteb/models/model_implementations/pylate_models.py (hunk 1 above)
Comment (NohTow) [reply mteb/models/model_implementations/pylate_models.py]: Good catch!
I swapped it everywhere but here :')
Updated!
